core/asset/asset.go

```diff
@@ -22,8 +22,10 @@ type Repository interface {
 	GetTypes(ctx context.Context, flt Filter) (map[Type]int, error)
 	Upsert(ctx context.Context, ast *Asset) (string, error)
 	UpsertPatch(ctx context.Context, ast *Asset, patchData map[string]interface{}) (string, error)
-	DeleteByID(ctx context.Context, id string) error
+	DeleteByID(ctx context.Context, id string) (string, error)
```

Comment:
instead introduce new method `SoftDelete*`, let's keep using the existing one, will override the behavior to soft delete, hard deletion will come later on